advisor-go/src/quixo/game/game.go

```diff
@@ -1,2 +1,167 @@
+package game
 
+import (
+	"fmt"
+	"strconv"
+)
 
+// Board with Grid and Player
+type Board struct {
+	Grid   [][]int
+	Player int
+}
+
+// Coords x and y
+type Coords struct {
+	X int
+	Y int
+}
+
+// Cube with Coords and value
+type Cube struct {
+	Coords Coords
+	Value  int
+}
+
+// GetBoard return a Board from grid and player
+func GetBoard(grid [][]int, player int) Board {
+	newBoard := Board{
+		Grid:   grid,
+		Player: player,
+	}
+	return newBoard
+}
+
+// GetMovablesCubes return an array of movables cube
+func GetMovablesCubes(board Board) []Cube {
+	grid := board.Grid
+	player := board.Player
+
+	movables := []Cube{}
+	for x := 0; x < len(grid); x++ {
+		for y := 0; y < len(grid[x]); y++ {
+			cube := Cube{
+				Coords: Coords{X: x, Y: y},
+				Value:  grid[x][y],
+			}
+			if isOutsideCube(cube.Coords, len(grid[x])) && isPlayerCube(cube.Value, player) {
+				movables = append(movables, cube)
+			}
+		}
+	}
+	return movables
+}
+
+// GetAvailablesDestinations return an array of destinations
+func GetAvailablesDestinations(grid [][]int, coords Coords) []Coords {
+	destinations := []Coords{}
+	x := coords.X
+	y := coords.Y
+	indexEnd := len(grid) - 1
+
+	if x == 0 || x == indexEnd {
+		if y != 0 {
+			destinations = append(destinations, Coords{X: x, Y: 0})
+		}
+		if y != indexEnd {
+			destinations = append(destinations, Coords{X: x, Y: indexEnd})
+		}
+		if x == 0 {
+			destinations = append(destinations, Coords{X: indexEnd, Y: y})
+		} else {
+			destinations = append(destinations, Coords{X: 0, Y: y})
+		}
+	} else if y == 0 || y == indexEnd {
+		if x != 0 {
+			destinations = append(destinations, Coords{X: 0, Y: y})
+		}
+		if x != indexEnd {
+			destinations = append(destinations, Coords{X: indexEnd, Y: y})
+		}
+		if y == 0 {
+			destinations = append(destinations, Coords{X: x, Y: indexEnd})
+		} else {
+			destinations = append(destinations, Coords{X: x, Y: 0})
+		}
+	}
+
+	return destinations
+}
+
+// MoveCube from coordsStart to coordsEnd
+func MoveCube(board Board, coordsStart Coords, coordsEnd Coords) [][]int {
+	newGrid := DuplicateGrid(board.Grid)
+	player := board.Player
+	if coordsStart.X == coordsEnd.X {
+		return shiftLine(newGrid, coordsStart.Y, coordsEnd.Y, coordsStart.X, player)
+	}
+	if coordsStart.Y == coordsEnd.Y {
+		flippedGrid := FlipRowsAndCols(newGrid)
+		flippedGrid = shiftLine(newGrid, coordsStart.X, coordsEnd.X, coordsStart.Y, player)
+		return FlipRowsAndCols(flippedGrid)
+	}
+	return nil
+}
+
+func isOutsideCube(coords Coords, size int) bool {
+	return coords.X == 0 || coords.Y == 0 || coords.X == size-1 || coords.Y == size-1
+}
+
+func isPlayerCube(cubeValue int, playerValue int) bool {
```

Comment:
I advise you to use the same order for the function name as for the parameters: `isPlayerCube` => `cubeBelongsToPlayer`